xarray/plot/utils.py

```diff
@@ -253,12 +254,67 @@ def _determine_cmap_params(plot_data, vmin=None, vmax=None, cmap=None,
                 levels=levels, norm=norm)
 
 
-def _infer_xy_labels(darray, x, y):
+def _infer_xy_labels_3d(darray, x, y, rgb):
+    """
+    Determine x and y labels for showing RGB images.
+
+    Attempts to infer which dimension is RGB/RGBA by size and order of dims.
```

Comment:
Rather than all this complex logic with warnings in ambiguous cases, why not always treat the last and/or remaining (after explicit x/y labels) dimension as RGB? I think that solves the convenience use cases, without hard to understand/predict inference logic.

Reply:
Because `some_dataset[['red', 'green', 'blue']].to_array().plot.imshow()` is *really tempting*.  The only difference is that if the last dimension is too large, we try the first dimension before resuming 'last dim is color' behaviour.

On reflection I think you're right though - so long as we can do the right thing when only one dim is the right size, a simpler heuristic for order preference makes good sense.  Will change.